### PR TITLE
refactor: flatten deps and improve repo folder

### PR DESCRIPTION
**Pull request by Yohe-Am**
- Renames `libs/` to `src/`.
- Moves `typegate/`, `meta-cli`, `typegraph/`, `meta-lsp/` to `src/`.
- Renames `dev/` to `tools/`
- Moves `website/` to `docs/metatype.dev`/
- Moves `src/typegate/tests` to `tests/`
- Moves `src/typegraph/deno/dev` to `tools/jsr/`
- Moves `src/typegraph/deno/sdk` to `src/typegraph/deno/`
- Renames `src/deno` to `src/mt_deno`
- Bumps deno to `1.46.1`
- Bumps rust toolchain to `1.80.1`
- Moves all rust dependencies to workspace section
- Moves `tools/task-*.ts` to `tools/task/*.ts`
- Moves `cliff.toml`, `Cross.toml`, `ruff.toml` to `tools/`
- Uses deno [workspaces](https://docs.deno.com/runtime/manual/basics/workspaces/) to organize `src/typegraph/deno`, `src/typegate`, `tools/`, `tests/` and more.
- Closes MET-607
- Updates poetry to 1.8.3
- Bumps metatype to version 0.4.9-rc1
- [x] Update CONTRIBUTING.md
- [x] Fix CI workflows
- [x] Fix Dockerfile
- [x] Fix all tests

#### Migration notes

- No end user changes required

- [x] The change comes with new or modified tests
- [ ] Hard-to-understand functions have explanatory comments
- [ ] End-user documentation is updated to reflect the change

### CODE REVIEW

**Review by Yohe-Am — COMMENTED** [review]
This PR is best viewed inside the vscode github pr extension.

**Review by Yohe-Am — COMMENTED** [review]
This is 99% renames actually.

**Comment by Yohe-Am** [comment]
Note, I've updated this to a release candidate version just to make sure the changes to the release pipeline are good before anything.

**Review by zifeo — COMMENTED** [review]
LGTM, Massive!

There are 2 files left under website, likely due to a clashing merge:
CleanShot 2024-09-01 at 22 28 57@2x

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -184,21 +184,21 @@ jobs:
...
-          cd typegraph/node
+          cd src/typegraph/node
           pnpm config set '//registry.npmjs.org/:_authToken' "${NODE_AUTH_TOKEN}"
           [ "$OVERWRITE" == 'true' ] \
             && (pnpm publish --no-git-checks --force || true) \
             || pnpm publish --no-git-checks
           cd -
 
-          cd typegraph/deno/sdk
+          cd src/typegraph/deno/
```
Comment (michael-0acf4) [inline .github/workflows/release.yml:201]: If I recall the additional `sdk` folder was to emphasize the `@typegraph` namespace and other eventual packages but I think right now removing it is fine.

**Review by michael-0acf4 — APPROVED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (Yohe-Am) [reply .github/workflows/release.yml:201]: Ahh. My bad. I was trying to normalize it against the node/python folder structures.

**Review by michael-0acf4 — APPROVED** [review]

**Pull request merged**